sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,293 @@
+# Authors: Andrea Bravi <a.bravi@uottawa.ca>
+#          Nikolay Mayorov <n59_ru@hotmail.com>
+# License: 3-clause BSD
+
+from __future__ import division
+
+import numpy as np
+from scipy.sparse import issparse
+from scipy.special import digamma
+
+from ..base import BaseEstimator
+from .base import SelectorMixin
+from ..metrics.cluster.supervised import mutual_info_score
+from ..utils.validation import check_X_y
+from ..neighbors import NearestNeighbors
+from ..preprocessing import scale
+
+
+def _compute_mi_cc(x, y, n_neighbors, scale_variables):
+    """Compute mutual information between two continuous variables.
+
+    Parameter `scale_variables` determine whether to scale variables to
+    the unit variance before computing mutual information. The theoretical
+    value of mutual information doesn't change by that transformation, but
+    in  practice it can be very important to do this scaling. Basically the
+    purpose of this flag is to disable unnecessary scaling in scenarios when it
+    was done in outer scope of this function.
+
+    For the details of the algorithm see references in
+    MaxRelevanceMinRedundancy class.
+    """
+    n_samples = x.size
+
+    if scale_variables:
+        x_std = np.std(x)
+        if x_std > 0:
+            x = x / x_std
+        y_std = np.std(y)
+        if y_std > 0:
+            y = y / y_std
+
+    x = x.reshape((-1, 1))
+    y = y.reshape((-1, 1))
+    xy = np.hstack((x, y))
+
+    nn = NearestNeighbors(metric='chebyshev', n_neighbors=n_neighbors)
+
+    nn.fit(xy)
+    radius = nn.kneighbors()[0]
```

Comment:
Right. That's kind of embarrassing because I wrote it :P (https://github.com/scikit-learn/scikit-learn/commit/9ed1df004a1d1db995e7b275f8ef9001bc3fa7b4)